jwst/jump/tests/test_detect_jumps.py

```diff
@@ -354,79 +351,78 @@ def test_crs_on_edge_with_neighbor_flagging(setup_inputs):
-    model1.data[0, 7, 5, 0] = 160.0
-    model1.data[0, 8, 5, 0] = 170.0
-    model1.data[0, 9, 5, 0] = 180.0
+    model.data[0, 0, 5, 0] = 15.0
+    model.data[0, 1, 5, 0] = 20.0
+    model.data[0, 2, 5, 0] = 25.0
+    model.data[0, 3, 5, 0] = 30.0
+    model.data[0, 4, 5, 0] = 35.0
+    model.data[0, 5, 5, 0] = 140.0
+    model.data[0, 6, 5, 0] = 150.0
+    model.data[0, 7, 5, 0] = 160.0
+    model.data[0, 8, 5, 0] = 170.0
+    model.data[0, 9, 5, 0] = 180.0
     # CR on last column
-    model1.data[0, 0, 15, 1027] = 15.0
-    model1.data[0, 1, 15, 1027] = 20.0
-    model1.data[0, 2, 15, 1027] = 25.0
-    model1.data[0, 3, 15, 1027] = 30.0
-    model1.data[0, 4, 15, 1027] = 35.0
-    model1.data[0, 5, 15, 1027] = 140.0
-    model1.data[0, 6, 15, 1027] = 150.0
-    model1.data[0, 7, 15, 1027] = 160.0
-    model1.data[0, 8, 15, 1027] = 170.0
-    model1.data[0, 9, 15, 1027] = 180.0
-    out_model = detect_jumps(model1, gain, rnModel, 4.0,  1, 200, 10, True)
+    model.data[0, 0, 15, -1] = 15.0
+    model.data[0, 1, 15, -1] = 20.0
+    model.data[0, 2, 15, -1] = 25.0
+    model.data[0, 3, 15, -1] = 30.0
+    model.data[0, 4, 15, -1] = 35.0
+    model.data[0, 5, 15, -1] = 140.0
+    model.data[0, 6, 15, -1] = 150.0
+    model.data[0, 7, 15, -1] = 160.0
+    model.data[0, 8, 15, -1] = 170.0
+    model.data[0, 9, 15, -1] = 180.0
+    out_model = detect_jumps(model, gain, rnoise, 4.0,  1, 200, 10, True)
     # flag CR and three neighbors of first row CR
     assert (4 == out_model.groupdq[0, 5, 0, 15])
     assert (4 == out_model.groupdq[0, 5, 1, 15])
     assert (4 == out_model.groupdq[0, 5, 0, 14])
     assert (4 == out_model.groupdq[0, 5, 0, 16])
     assert (out_model.groupdq[0, 5, -1, 15] == 0) # The one not to flag
     # flag CR and three neighbors of last row CR
-    assert (4 == out_model.groupdq[0, 5, 1023, 5])
-    assert (4 == out_model.groupdq[0, 5, 1022, 5])
-    assert (4 == out_model.groupdq[0, 5, 1023, 4])
-    assert (4 == out_model.groupdq[0, 5, 1023, 6])
+    assert (4 == out_model.groupdq[0, 5, -1, 5])
+    assert (4 == out_model.groupdq[0, 5, -2, 5])
+    assert (4 == out_model.groupdq[0, 5, -1, 4])
+    assert (4 == out_model.groupdq[0, 5, -1, 6])
     # flag CR and three neighbors of first column CR
     assert (4 == out_model.groupdq[0, 5, 5, 0])
     assert (4 == out_model.groupdq[0, 5, 6, 0])
     assert (4 == out_model.groupdq[0, 5, 4, 0])
     assert (4 == out_model.groupdq[0, 5, 5, 1])
     assert (out_model.groupdq[0, 5, 5, -1] == 0)# The one not to flag
     # flag CR and three neighbors of last column CR
-    assert (4 == out_model.groupdq[0, 5, 15, 1027])
-    assert (4 == out_model.groupdq[0, 5, 15, 1026])
-    assert (4 == out_model.groupdq[0, 5, 16, 1027])
-    assert (4 == out_model.groupdq[0, 5, 14, 1027])
+    assert (4 == out_model.groupdq[0, 5, 15, -1])
+    assert (4 == out_model.groupdq[0, 5, 15, -2])
+    assert (4 == out_model.groupdq[0, 5, 16, -1])
+    assert (4 == out_model.groupdq[0, 5, 14, -1])
```

Comment:
It's not clear to me the original test was testing this properly.  The input array size is

```
nrows = 1024
ncols = 1032
```

but the checks were actually not being made at the edge on columns 1030 and 1031, but in columns 1026 and 1027.  I suspect these are reference pixels, but these tests don't know anything about reference pixels.

For the rows, the check was being done correctly, at row 1022 and 1023.

Regardless, both tests pass in the new version of the test.

Reply:
Yes, the last few cols are ref pixels, hence the tests were being done in a few cols from the edge. But then that means they aren't really "edge" pixels, because the data array that's being examined by the jump step includes the extra cols of ref pixels.